### PR TITLE
WIP: federated e2e runner

### PR DESCRIPTION
**Pull request by justinsb**
WIP on the runner component of federated e2e testing.  Lots of problems still to solve, but a PR is a nice way to discuss & collaborate IMO.

Runs the e2e tests in a Docker container, and uploads results to a GCS bucket.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/477)

### CODE REVIEW

File: federated_runner/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM debian:jessie
+
+RUN apt-get update && apt-get install --yes curl python-pip openssh-client
+RUN pip install awscli
+
+RUN curl https://sdk.cloud.google.com | bash
+
+RUN curl https://storage.googleapis.com/golang/go1.5.3.linux-amd64.tar.gz | tar -C /usr/local -xzf -
```
Comment (ixdy) [inline federated_runner/Dockerfile]: is there a reason not to just base this on the golang image?

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by grodrigues3** [comment]
[APPROVALNOTIFIER] The Following OWNERS Files Need Approval:

- **/**

We suggest the following people:
cc @lavalamp 
 You can indicate your approval by writing `/approve` in a comment
 You can cancel your approval by writing `/approve cancel`in a comment

**Comment by spiffxp** [comment]
/close
Educated guess that this isn't getting done.  Most of the underlying code has since moved on.  There are efforts underway in kubernetes/test-infra to do something like this via bootstrap and sidecar containers

**Pull request closed without merging**